src/main/java/core/stack/MateStack.java

```diff
@@ -1,19 +1,55 @@
 package core.stack;
 
+import java.util.EmptyStackException;
+
 public class MateStack<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private Object[] elementData;
+    private int size;
+
+    public MateStack() {
+        elementData = new Object[DEFAULT_CAPACITY];
+    }
+
     public void push(T value) {
+        ensureCapacity();
+        elementData[size] = value;
+        size++;
+    }
```

Comment:
Consider checking for null values before adding them to the stack. Allowing nulls may be a design choice, but if not intentional, it could lead to unexpected behavior when using the `peek` or `pop` methods.